src/libcollections/vec.rs

```diff
@@ -1106,12 +1106,36 @@ impl<T: Clone> Vec<T> {
         let len = self.len();
 
         if new_len > len {
-            self.extend(repeat(value).take(new_len - len));
+            self.extend_elem(new_len - len, value);
         } else {
             self.truncate(new_len);
         }
     }
 
+    /// Extend the vector by `n` additional clones of `value`.
+    #[inline(always)]
```

Comment:
Can you omit this `#[inline(always)]` tag? This seems like a fairly significant chunk of code to *always* inline. I would think the tag could be omitted entirely as well (this is already generic)